shopBuddy/shopBuddy/ItemDetailViewController.m

```diff
@@ -50,8 +52,11 @@ - (void)_callAPI {
     [[APIManager shared] getItemWithBarcode:self.barcode completion:^(Item *item, NSError *error) {
         if (item) {
             __strong __typeof(weakSelf)strongSelf = weakSelf;
-            strongSelf.item = item;
-            [weakSelf _populateView];
+            if(weakSelf)
+            {
+                strongSelf->_item = item;
+                [weakSelf _populateView];
+            }
```

Comment:
Same case, see comment above.
Another option to avoid this hassle is to extract to a private method and use weak ref instead.
```
    [[APIManager shared] getItemWithBarcode:self.barcode completion:^(Item *item, NSError *error) {
                    [weakSelf _handleBarcodeResponse:item];
    }];
....


- (void)_handleBarcodeResponse:(Item *)item {
                self._item = item;
                [self _populateView];
}
```